src/sentry/search/snuba/executors.py

```diff
@@ -0,0 +1,734 @@
+    """
+    Finds the value of a search filter with the passed name and operator. If
+    multiple values are found, returns the most restrictive value
+    :param search_filters: collection of `SearchFilter` objects
+    :param name: Name of the field to find
+    :param operator: '<' or '>'
+    :return: The value of the field if found, else None
+    """
+    assert operator in ("<", ">")
+    comparator = max if operator.startswith(">") else min
+    found_val = None
+    for search_filter in search_filters:
+        # Note that we check operator with `startswith` here so that we handle
+        # <, <=, >, >=
+        if search_filter.key.name == name and search_filter.operator.startswith(operator):
+            val = search_filter.value.raw_value
+            found_val = comparator(val, found_val) if found_val else val
+    return found_val
+
+
+@six.add_metaclass(ABCMeta)
+class AbstractQueryExecutor:
+    """This class serves as a template for Query Executors.
+    We subclass it in order to implement query methods (we use it to implement two classes: joined Postgres+Snuba queries, and Snuba only queries)
+    It's used to keep the query logic out of the actual search backend,
+    which can now just build query parameters and use the appropriate query executor to run the query
+    """
+
+    EMPTY_RESULT = Paginator(Group.objects.none()).get_result()
+    TABLE_ALIAS = ""
+
+    @abstractmethod
+    def query(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def snuba_search(
+        self,
+        start,
+        end,
+        project_ids,
+        environment_ids,
+        sort_field,
+        cursor=None,
+        group_ids=None,
+        limit=None,
+        offset=0,
+        get_sample=False,
+        search_filters=None,
+    ):
+        """
+        Returns a tuple of:
+        * a sorted list of (group_id, group_score) tuples sorted descending by score,
+        * the count of total results (rows) available for this query.
+        """
+        if not self.QUERY_DATASET:
+            raise NotImplementedError
+
+        filters = {"project_id": project_ids}
+
+        if environment_ids is not None:
+            filters["environment"] = environment_ids
+
+        if group_ids:
+            filters[self.TABLE_ALIAS + "issue"] = sorted(group_ids)
+
+        conditions = []
+        having = []
+        for search_filter in search_filters:
+            if (
+                # Don't filter on issue fields here, they're not available
+                search_filter.key.name in self.issue_only_fields
+                or
+                # We special case date
+                search_filter.key.name == "date"
+            ):
+                continue
+            converted_filter = convert_search_filter_to_snuba_query(search_filter)
+            converted_filter, converted_name = self.modify_converted_filter(
+                search_filter, converted_filter, project_ids, environment_ids
```

Comment:
I changed it to transform instead of modify, but I realize that doesn't help much.

I added a doc string to the function to help more, instead.

Will consider better names.